src/storage/cloud_storage.rs

```diff
@@ -41,6 +41,27 @@ struct Item {
     size: String,
 }
 
+// JSON error response format:
+// https://cloud.google.com/storage/docs/json_api/v1/status-codes
+#[derive(Deserialize, Debug)]
+struct ErrorBodyErrors {
+    domain: String,
+    reason: String,
+    message: String,
+}
+
+#[derive(Deserialize, Debug)]
+struct ErrorBodyError {
+    errors: Vec<ErrorBodyErrors>,
+    code: u32,
+    message: String,
+}
+
+#[derive(Deserialize, Debug)]
+struct ErrorBody {
```

Comment:
I can think of this as the ResponseBody too, which in case of an error should hold the ```error``` part, otherwise the ```items``` and ```prefixes```. Was there a reason why you did not add an ```Optional<ErrorBodyError>``` to the ```ResponseBody```?